src/Compilers/CSharp/Test/Semantic/Semantics/SimpleProgramsTests.cs

```diff
@@ -40,22 +42,47 @@ public void Simple_02()
 ";
 
             var comp = CreateCompilation(text, options: TestOptions.DebugExe, parseOptions: DefaultParseOptions);
-            Assert.Equal("System.Threading.Tasks.Task $Program.$Main()", SimpleProgramNamedTypeSymbol.GetSimpleProgramEntryPoint(comp).ToTestDisplayString());
+            Assert.Equal("System.Threading.Tasks.Task", SimpleProgramNamedTypeSymbol.GetSimpleProgramEntryPoint(comp).ReturnType.ToTestDisplayString());
             CompileAndVerify(comp, expectedOutput: "hello async main");
         }
 
         [Fact]
         public void Simple_03()
         {
-            var text1 = @"System.Console.WriteLine(""1"");";
-            var text2 = @"System.Console.WriteLine(""2"");";
+            var text1 = @"
+System.Console.Write(""1"");
+";
+            var text2 = @"
+//
+System.Console.Write(""2"");
+System.Console.WriteLine();
+System.Console.WriteLine();
+";
+            var text3 = @"
+//
+//
+System.Console.Write(""3"");
+System.Console.WriteLine();
+System.Console.WriteLine();
+";
 
             var comp = CreateCompilation(new[] { text1, text2 }, options: TestOptions.DebugExe, parseOptions: DefaultParseOptions);
 
             comp.VerifyDiagnostics(
-                // (1,1): error CS8751: Internal error in the C# compiler.
-                // System.Console.WriteLine("2");
-                Diagnostic(ErrorCode.ERR_InternalError, @"System.Console.WriteLine(""2"");").WithLocation(1, 1)
+                // (3,1): error CS9001: In all but one compilation unit the top-level statements must all be local function declarations.
```

Comment:
I think this message could be refined to be easier for new language learners for understand, for instance by using the term "file" instead of "compilation unit". Something to consider for a future PR.